src/cdomains/intDomain.ml

```diff
@@ -1317,7 +1317,16 @@ struct
           let r = size ik in
           (* Perform a wrap-around for unsigned values and for signed values (if configured). *)
           let mapped_excl = S.map (fun excl -> BigInt.cast_to ik excl) s in
-          `Excluded (mapped_excl, r)
+          match ik with
+          | IBool ->
+            begin match S.mem BigInt.zero mapped_excl, S.mem BigInt.one mapped_excl with
+              | false, false -> `Excluded (mapped_excl, r) (* Not {} -> Not {} *)
+              | true, false -> `Definite BigInt.one (* Not {0} -> 1 *)
+              | false, true -> `Definite BigInt.zero (* Not {1} -> 0 *)
+              | true, true -> `Bot (* Not {0, 1} -> bot *)
```

Comment:
Are we sure this is only ever called with a value that is normalized before, and not some normal integer that was computed somewhere else where `Not {0,1}` would be truthy?

Reply:
Not sure what you mean. The `mapped_excl` above normalizes the values to the ikind and this is only done for `IBool`, which only has the values 0 and 1 after normalizing.